src/test/kotlin/org/pkl/lsp/ParserTest.kt

```diff
@@ -196,6 +196,28 @@ class ParserTest {
     assertThat(prop.parsedComment).isEqualTo("A doc\ncomment")
   }
 
+  @Test
+  fun `when generators`() {
+    val code =
+      """
+      foo {
+        when (true) {
+          bar = 1
+        } else {
+          bar = 2
+        }
+      }
+    """
+        .trimIndent()
+    val mod = parse(code)
+    val prop = mod.children[0]
+    assertThat(prop).isInstanceOf(PklClassProperty::class.java)
+    prop as PklClassProperty
+    val whenGenerator = prop.objectBody!!.members.first() as PklWhenGenerator
+    assertThat(whenGenerator.thenBody).isNotNull
+    assertThat(whenGenerator.elseBody).isNotNull
```

Comment:
```suggestion
    assertThat(whenGenerator.thenBody).isInstanceOf(PklObjectBody::class.java)
    assertThat(whenGenerator.elseBody).isInstanceOf(PklObjectBody::class.java)
```
nit: a bit stronger check.

Reply:
This is already ensured by the compiler!